### PR TITLE
use sub folder when caching data

### PR DESCRIPTION
**Pull request by TonioF**
As we talked about.

### CODE REVIEW

File: cate/core/ds.py [hunk 1]
```diff
@@ -485,6 +487,10 @@ def make_local(data: Any,
         while local_store.has_data(local_name):
             i += 1
             local_name = f'local.{orig_dataset_name}.{i}{extension}'
+    if '/' not in local_name:
+        local_name_prefix = f'cate-local/{data_store_id}/' \
+            if data_store_id is not None else 'cate-local/'
```
Comment (forman) [inline cate/core/ds.py:492]:
```suggestion
        local_name_prefix = f'{CATE_LOCAL_DIR_NAME}/{data_store_id}/' \
            if data_store_id is not None else f'{CATE_LOCAL_DIR_NAME}/'
```

**Review by forman — CHANGES_REQUESTED** [review]
Also, please add a changelog entry.

**Review by TejasMorbagal — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 * The "local" data store is now configured to use the current 
   working directory as root.
+  
+* Cached data is now written into a sub-folder `cate-local/{data_store_id}` 
+  within the localc data store root unless another folder is specified.
```
Comment (forman) [inline CHANGES.md:16]:
```suggestion
  within the Cate server's root path unless another folder is specified. 
```

File: cate/core/ds.py [hunk 3]
```diff
@@ -110,16 +110,8 @@
...
-    r'^(?:http|ftp)s?://'  # http:// or https://
-    r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|'  # domain
-    r'localhost|'  # localhost...
-    r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'  # ...or ip
-    r'(?::\d+)?'  # optional port
-    r'(?:/?|[/?]\S+)$', re.IGNORECASE)
-
 _LOG = logging.getLogger('cate')
-
+CATE_LOCAL_DIR_NAME = 'cate-local'
```
Comment (forman) [inline cate/core/ds.py:114]: Ideally, this should go into `defaults.py` like all the other global constants.

**Review by forman — COMMENTED** [review]
Ok! You may consider my suggestions.

**Comment by TonioF** [comment]
Accepted suggestions, I merge.

**Pull request merged**